File: java/Figuras/src/main/java/org/example/Cuadrado.java
package org.example;

/**
 * Clase Cuadrado - Figura bidimensional
 */
public class Cuadrado extends Figura2d {
    
    private double lado;
    
    /**
     * Constructor de la clase Cuadrado
     * @param lado Longitud del lado del cuadrado
     * @param idFigura Identificador único de la figura
     * @throws IllegalArgumentException Si el lado es menor o igual a cero
     */
    public Cuadrado(double lado, int idFigura) {
        super("Cuadrado", idFigura);
        if (lado <= 0) {
            throw new IllegalArgumentException("El lado debe ser mayor que cero");
        }
        this.lado = lado;
    }
    
    /**
     * Calcula el área del cuadrado
     * @return El área del cuadrado
     */
    @Override
    public double calcularArea() {
        return this.lado * this.lado;
    }
    
    /**
     * Calcula el perímetro del cuadrado
     * @return El perímetro del cuadrado
     */
    @Override
    public double calcularPerimetro() {
        return 4 * this.lado;
    }
    
    /**
     * Obtiene la longitud del lado del cuadrado
     * @return La longitud del lado
     */
    public double getLado() {
        return this.lado;
    }
    
    /**
     * Establece la longitud del lado del cuadrado
     * @param lado La nueva longitud del lado
     * @throws IllegalArgumentException Si el lado es menor o igual a cero
     */
    public void setLado(double lado) {
        if (lado <= 0) {
            throw new IllegalArgumentException("El lado debe ser mayor que cero");
        }
        this.lado = lado;
    }
    
    /**
     * Representación en cadena del cuadrado
     * @return Representación del cuadrado
     */
    @Override
    public String toString() {
        return String.format("Cuadrado(id=%d, lado=%.2f)", getId(), lado);
    }
}

File: python/Figuras/cuadrado.py
"""
Clase Cuadrado - Figura bidimensional
"""
# pylint: disable=invalid-name
from Figura2d import Figura2d

class Cuadrado(Figura2d):
    """Clase que representa un cuadrado"""

    def __init__(self, lado: float, id_figura: int):
        """
        Constructor de la clase Cuadrado

        Args:
            lado (float): Longitud del lado del cuadrado
            id_figura (int): Identificador único de la figura
        
        Raises:
            ValueError: Si el lado es menor o igual a cero
        """
        if lado <= 0:
            raise ValueError("El lado debe ser mayor que cero")
        super().__init__("Cuadrado", id_figura)
        self._lado = lado

    def calcular_area(self) -> float:
        """Calcula el área del cuadrado"""
        return self._lado ** 2

    def calcular_perimetro(self) -> float:
        """Calcula el perímetro del cuadrado"""
        return 4 * self._lado

    def get_lado(self) -> float:
        """Obtiene la longitud del lado del cuadrado"""
        return self._lado

    def set_lado(self, lado: float) -> None:
        """Establece la longitud del lado del cuadrado"""
        if lado <= 0:
            raise ValueError("El lado debe ser mayor que cero")
        self._lado = lado

    def __str__(self) -> str:
        """Representación en cadena del cuadrado"""
        return f"Cuadrado(id={self.get_id()}, lado={self._lado:.2f})"

    def __repr__(self) -> str:
        """Representación técnica del cuadrado"""
        return f"Cuadrado(lado={self._lado}, id_figura={self.get_id()})"
